### PR TITLE
feat: new id component

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- new ID component

## Test Plan

- tbd

## Related PRs and Issues

- #327 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

✅

### CODE REVIEW

File: src/lib/components/id.svelte [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            isOverflowing(element, iterator + 1);
+        }
+    }
+
+    $: isOverflowing(element);
+</script>
+
+<Copy {value} {event}>
+    <div class="interactive-text-output" style:min-inline-size="0" style:display="inline-flex">
+        <span style:white-space="nowrap" class="text u-line-height-1-5" bind:this={element}>
```
Comment (ArmanNik) [inline src/lib/components/id.svelte:36]: should we add data-private here by default?

File: src/routes/console/project-[project]/auth/user-[user]/header.svelte [hunk 2]
```diff
@@ -39,12 +38,7 @@
         <CoverTitle href={`/console/project-${projectId}/auth`}>
             {$user.name ? $user.name : '-'}
         </CoverTitle>
-        <Copy value={$user.$id} event="user">
-            <Pill button>
-                <span class="icon-duplicate" aria-hidden="true" />
-                User ID
-            </Pill>
-        </Copy>
+        <ID value={$user.$id} event="user">{$user.$id}</ID>
```
Comment (ArmanNik) [inline src/routes/console/project-[project]/auth/user-[user]/header.svelte:41]: Why we have `x?.$id` in some places but not here?

File: src/routes/console/project-[project]/databases/table.svelte [hunk 3]
```diff
@@ -36,21 +35,18 @@
...
                             {#key $columns}
-                                <TableCell title={column.title}>
-                                    <Copy value={database.$id}>
-                                        <Pill button trim>
-                                            <span class="icon-duplicate" aria-hidden="true" />
-                                            <span class="text u-trim">{database.$id}</span>
-                                        </Pill>
-                                    </Copy>
+                                <TableCell width={column.width} title={column.title}>
+                                    <ID value={database.$id}>
```
Comment (ArmanNik) [inline src/routes/console/project-[project]/databases/table.svelte:39]: Broken on mobile

image

**Review by ArmanNik — CHANGES_REQUESTED** [review]

File: src/routes/console/project-[project]/databases/table.svelte (hunk 3 above)
Comment (TorstenDittmann) [reply src/routes/console/project-[project]/databases/table.svelte:39]: fixed

File: src/routes/console/project-[project]/auth/user-[user]/header.svelte (hunk 2 above)
Comment (TorstenDittmann) [reply src/routes/console/project-[project]/auth/user-[user]/header.svelte:41]: not sure, but the data should always be here 👍🏻

File: src/lib/components/id.svelte (hunk 1 above)
Comment (TorstenDittmann) [reply src/lib/components/id.svelte]: added

File: src/lib/components/id.svelte [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            },
+            destroy() {
+                window.removeEventListener('resize', checkOverflow);
+            }
+        };
+    }
+</script>
+
+<Copy {value} {event}>
+    <div class="interactive-text-output" style:min-inline-size="0" style:display="inline-flex">
```
Comment (ArmanNik) [inline src/lib/components/id.svelte:49]: This is missing the class `is-buttons-on-top` :+1:

File: src/lib/components/copy.svelte [hunk 5]
```diff
@@ -33,6 +33,7 @@
 </script>
 
 <span
+    data-private
```
Comment (TGlide) [inline src/lib/components/copy.svelte:36]: What is this for?

File: src/lib/components/index.ts [hunk 6]
```diff
@@ -53,3 +53,4 @@ export { default as Limit } from './limit.svelte';
 export { default as PaginationWithLimit } from './paginationWithLimit.svelte';
 export { default as ClickableList } from './clickableList.svelte';
 export { default as ClickableListItem } from './clickableListItem.svelte';
+export { default as ID } from './id.svelte';
```
Comment (TGlide) [inline src/lib/components/index.ts:56]: Nitpicking, but I find it weird that it's ID and not Id

**Review by TGlide — APPROVED** [review]

File: src/lib/components/copy.svelte (hunk 5 above)
Comment (TorstenDittmann) [reply src/lib/components/copy.svelte:36]: hides informations on logrocket

**Pull request merged**